scipy/optimize/_numdiff.py

```diff
@@ -177,8 +176,8 @@ def group_columns(A, order=0):
 
 
 def approx_derivative(fun, x0, method='3-point', rel_step=None, f0=None,
-                      bounds=(-np.inf, np.inf), sparsity=None, args=(),
-                      kwargs={}):
+                      bounds=(-np.inf, np.inf), sparsity=None,
+                      as_linear_operator=False, args=(), kwargs={}):
```

Comment:
`as_linear_operator` should be last in the keyword arguments, to maintain the same order.

Reply:
I think it's better to order it like this, it is not a public function yet. But we should make it public.

Reply:
I think I agree with Nikolay here. We do not have to care much with compatibility, because the function has not been made public yet